backend/src/graphql/resolver/ContributionResolver.ts

```diff
@@ -425,40 +424,30 @@ export class ContributionResolver {
   }
 
   @Authorized([RIGHTS.LIST_UNCONFIRMED_CONTRIBUTIONS])
-  @Query(() => [UnconfirmedContribution])
-  async listUnconfirmedContributions(@Ctx() context: Context): Promise<UnconfirmedContribution[]> {
-    const clientTimezoneOffset = getClientTimezoneOffset(context)
-    const contributions = await getConnection()
-      .createQueryBuilder()
-      .select('c')
-      .from(DbContribution, 'c')
-      .leftJoinAndSelect('c.messages', 'm')
-      .where({ confirmedAt: IsNull() })
-      .andWhere({ deniedAt: IsNull() })
-      .getMany()
-
-    if (contributions.length === 0) {
-      return []
-    }
-
-    const userIds = contributions.map((p) => p.userId)
-    const userCreations = await getUserCreations(userIds, clientTimezoneOffset)
-    const users = await DbUser.find({
-      where: { id: In(userIds) },
+  @Query(() => ContributionListResult) // [UnconfirmedContribution]
+  async adminListAllContributions(
+    @Args()
+    { currentPage = 1, pageSize = 3, order = Order.DESC }: Paginated,
+    @Arg('statusFilter', () => [ContributionStatus], { nullable: true })
+    statusFilter?: ContributionStatus[],
+  ): Promise<ContributionListResult> {
+    const [dbContributions, count] = await DbContribution.findAndCount({
```

Comment:
Put this in a separated function in the `utils` folder. Use that function as well for the frontend call

Reply:
Line 179